### PR TITLE
Data ingestion

### PR DESCRIPTION
**Pull request by vianamp**
We are sampling data from LabKey on the fly instead of loading a local CSV file :)

### CODE REVIEW

File: mti_nma/steps/singlecell/singlecell.py [hunk 1]
```diff
@@ -34,62 +35,102 @@ def __init__(
...
         Parameters
         ----------
+        cell_line_id: str
+            Name of the cell line where nuclei are going to be sampled from
+            AICS-13 = Lamin
+
         nsamples: int
             Number of cells to sample randomly (but with set seed) from dataset
-        """
+        '''
```
Comment (jcass11) [inline mti_nma/steps/singlecell/singlecell.py]: I really appreciate your detailed function docstrings - really nice!

File: mti_nma/steps/singlecell/singlecell.py [hunk 2]
```diff
@@ -34,62 +35,102 @@ def __init__(
...
+
+            df = df.sample(n=nsamples)
+
+            for fov_id in tqdm(df.index[:1]):
+
+                sx = df.PixelScaleX[fov_id]
+                sy = df.PixelScaleY[fov_id]
+                sz = df.PixelScaleZ[fov_id]
+
+                # C=-2 may not return the DNA channel for other cell lines. Need validation.
```
Comment (jcass11) [inline mti_nma/steps/singlecell/singlecell.py]: Is this something we want to check post PR-merge?

File: mti_nma/steps/singlecell/singlecell.py [hunk 3]
```diff
@@ -34,62 +35,102 @@ def __init__(
...
+                seg = AICSImage(df.ReadPathSeg[fov_id]).get_image_data('ZYX', S=0, T=0, C= 0)
+
+                # Select one label from seg image at random
+                obj_label = np.random.randint(low=1, high=1+seg.max())
+
+                raw, seg = crop_object(
+                    raw = raw,
+                    seg = seg,
+                    obj_label = obj_label,
+                    isotropic = (sx,sy,sz))
```
Comment (jcass11) [inline mti_nma/steps/singlecell/singlecell.py]: Is this setting all seg images to be the same dimensions?

Comment (jcass11) [reply mti_nma/steps/singlecell/singlecell.py]: Based on the docstring and the code I'm pretty sure it is but want to be 100% sure I'm following

File: mti_nma/steps/singlecell/singlecell.py [hunk 4]
```diff
@@ -34,62 +35,102 @@ def __init__(
...
+
+                # Create an unique id for this object
+                cell_id = uuid.uuid4().hex[:8]
+
+                # Save images
+                writer = writers.OmeTiffWriter(self.step_local_staging_dir.as_posix() + f'/{cell_id}.raw.tif')
+                writer.save(raw, dimension_order='ZYX')
+
+                writer = writers.OmeTiffWriter(self.step_local_staging_dir.as_posix() + f'/{cell_id}.seg.tif')
+                writer.save(seg, dimension_order='ZYX')
```
Comment (jcass11) [inline mti_nma/steps/singlecell/singlecell.py]: Maybe throw a comment in the function docstring indicating that the channel order of the stored segs is ZYX

File: mti_nma/steps/singlecell/singlecell_utils.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    df_seg = df_seg.rename(columns={'FOVId': 'FOVIdList', 'ReadPath': 'ReadPathSeg'})
+
+    df_raw = pd.DataFrame(query_raw)
+    df_raw = df_raw.rename(columns={'SourceImageFileId/LocalFilePath': 'ReadPathRaw'})
+
+    # Merging raw and seg tables
+
+    for index in df_seg.index:
+        fov_id = df_seg.FOVIdList[index]
+        if len(fov_id):
```
Comment (jcass11) [inline mti_nma/steps/singlecell/singlecell_utils.py]: Sorry - I don't quite understand what this if/block is doing

File: mti_nma/steps/singlecell/singlecell_utils.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
+import numpy as np
+import pandas as pd
+import labkey as lk
+from lkaccess import LabKey as lka
+from skimage import transform as sktrans
+
+def query_data_from_labkey(CellLineId):
```
Comment (jcass11) [inline mti_nma/steps/singlecell/singlecell_utils.py]: Could you add docstrings for both of these functions? I've been using a template style for docstrings if you want it to keep consistent or you can do whatever you like.

File: mti_nma/steps/singlecell/singlecell.py (hunk 2 above)
Comment (vianamp) [reply mti_nma/steps/singlecell/singlecell.py]: We may ask Jamie whether this is always true or not. Or maybe a better idea is to use `get_channel_names` to look for the DNA channel `H3342`. Then we can be sure we will always be looking at the correct channel. I am going to create an issue for this one.

File: mti_nma/steps/singlecell/singlecell.py (hunk 3 above)
Comment (vianamp) [reply mti_nma/steps/singlecell/singlecell.py]: Yes. All seg and raw images are going to have same scale in x, y, and z (scale=0.135um).

File: mti_nma/steps/singlecell/singlecell_utils.py (hunk 5 above)
Comment (vianamp) [reply mti_nma/steps/singlecell/singlecell_utils.py]: Sorry. This should have been better explained. The problem here is that some of the FOVs on labkey do not have segmentation and therefore the list FOVIdList is empty. I added a comment to address that.

File: mti_nma/steps/singlecell/singlecell_utils.py (hunk 6 above)
Comment (vianamp) [reply mti_nma/steps/singlecell/singlecell_utils.py]: Will do.

**Pull request merged**